scripts/build/autogen_ros_version_src.py

```diff
@@ -160,13 +169,19 @@ def get_desired_state(checkout_dir, autogen_dir, use_ros1):
     mapping paths under autogen_dir to their desired type and content.
     """
     result = {}
+    ignore_marker = "ROS1_IGNORE" if use_ros1 else "ROS2_IGNORE"
     for dir_path, dir_names, file_names in os.walk(checkout_dir, followlinks=True):
-        if dir_path == checkout_dir and ".git" in dir_names:
-            dir_names.remove(".git")  # skip .git subfolder
+        if ignore_marker in file_names:
+            # if folder contains ignore_marker file, ignore it and all of its
+            # children
+            dir_names[:] = []
+            continue
+
+        if ".git" in dir_names:
```

Comment:
Hm. I had a good reason to ignore the `.git` folder, which is that it's an error to try to use git inside the symlinked copy. I'm not sure what `git status` would show exactly, but there would be a bunch of bogus "changes" because the contents of the symlinked output are not the same as the source checkout by design, and it would break things if you accidentally commit some of these changes. So I thought it would be cleaner to leave out the `.git` folder so git would give you a helpful error that you are not inside a working copy.

There may be other files or folders where we don't benefit from symlinking them in the copy, but there's also no particular downside to having them there. In that case, I would say let's keep the script's ignore configuration simple (hard code only `.git` and the special ignore markers) even if it means the script does a bit of unnecessary extra work to link unneeded files.

But if you have an example in mind of other ignore patterns that would provide a significant benefit, that could change my opinion.